R/bayesiannetworkanalysis.R

```diff
@@ -0,0 +1,1297 @@
+                 factorLevels.amount = "< 2",
+                 exitAnalysisIfErrors = TRUE)
+      dataset[[.v(options[["groupingVariable"]])]] <- groupingVariable
+      groupingVariable <- options[["groupingVariable"]]
+    }
+  }
+}
+
+.bayesianNetworkAnalysisRun <- function(mainContainer, dataset, options) {
+  
+  # List that contains state or is empty:
+  networkList <- list(
+    network    = mainContainer[["networkState"]]$object,
+    centrality = mainContainer[["centralityState"]]$object,
+    layout     = mainContainer[["layoutState"]]$object
+  )
+  
+  if (length(options[["variables"]]) <= 2L)
+    return(networkList)
+  
+  if (is.null(networkList[["network"]]))
+    tryCatch(
+      networkList[["network"]] <- .bayesianNetworkAnalysisComputeNetworks(options, dataset),
+      error = function(e) {
+        mainContainer$setError(.extractErrorMessage(e))
+      }
+    )
+  
+  if (!mainContainer$getError() && !is.null(networkList[["network"]])) { 
+    if (is.null(networkList[["layout"]]))
+      networkList[["layout"]] <- .bayesianNetworkAnalysisComputeLayout(networkList[["network"]], dataset, options)
+    
+    if (is.null(networkList[["centrality"]]) && (options[["tableCentrality"]] || options[["plotCentrality"]]))
+      networkList[["centrality"]] <- .bayesianNetworkAnalysisComputeCentrality(networkList[["network"]])
+    
+    names(networkList[["network"]]) <- names(dataset) # <- names(networkList[["centrality"]])
+    
+    mainContainer[["networkState"]]    <- createJaspState(networkList[["network"]])
+    mainContainer[["centralityState"]] <- createJaspState(networkList[["centrality"]], dependencies = c("maxEdgeStrength", "minEdgeStrength"))
+    mainContainer[["layoutState"]]     <- createJaspState(networkList[["layout"]], 
+                                                          dependencies = c("layout", "repulsion", "layoutX", "layoutY"))
+    
+    #.networkAnalysisSaveLayout(mainContainer, options, networkList[["layout"]])
+  }
+  
+  return(networkList)
+}
+
+.bayesianNetworkAnalysisComputeCentrality <- function(networks) {
+  
+  centralities <- vector("list", length(networks))
+  for (nw in seq_along(networks)) {
+    
+    network <- networks[[nw]]
+    
+    # add measures 
+    centralitySamples <- centrality(network = network)
+    
+    nSamples <- nrow(network$samplesPosterior)
+    
+    value <- apply(centralitySamples[, 3:(nSamples+2)], MARGIN = 1, mean)
+    centralityHDIintervals <- apply(centralitySamples[, 3:(nSamples+2)], MARGIN = 1, 
+                                    FUN = HDInterval::hdi, allowSplit = FALSE)
+    
+    centralitySummary <- cbind(centralitySamples[, 1:2], value, t(centralityHDIintervals))
+    
+    centralities[[nw]] <- centralitySummary
+
+  }
+  
+  return(centralities)
+}
+
+.bayesianNetworkAnalysisComputeLayout <- function(networks, dataset, options) {
+
+  # Reformat networks to fit averageLayout:
+  weightMatrices <- list()
+  for (i in 1:length(networks)) {
```

Comment:
```suggestion
  for (i in seq_along(networks)) {
```
now I hope you didn't copy that loop from code I wrote :grin: